src/main/java/io/getunleash/impactmetrics/HistogramImpl.java

```diff
@@ -0,0 +1,104 @@
+package io.getunleash.impactmetrics;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+class HistogramImpl implements Histogram {
+    private final BucketMetricOptions options;
+    private final List<Double> buckets;
+    private final ConcurrentHashMap<String, HistogramData> values = new ConcurrentHashMap<>();
+
+    private static final List<Double> DEFAULT_BUCKETS =
+            List.of(0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0);
+
+    HistogramImpl(BucketMetricOptions options) {
+        this.options = options;
+        List<Double> inputBuckets = options.getBuckets();
+        if (inputBuckets == null || inputBuckets.isEmpty()) {
+            inputBuckets = DEFAULT_BUCKETS;
+        }
+
+        List<Double> sorted =
+                inputBuckets.stream()
+                        .filter(d -> d != Double.POSITIVE_INFINITY)
+                        .distinct()
+                        .sorted()
+                        .collect(Collectors.toList());
+        sorted.add(Double.POSITIVE_INFINITY);
```

Comment:
In java there is no such thing as `+Inf` as string and hard to model. In java there is `POSITIVE_INFINITY` and we can later change to `+Inf` if need to send to Unleash.